### PR TITLE
[C-2494] Refactor drag-drop to typescript + performance

### PR DESCRIPTION
**Pull request by dylanjeffers**
### Description

Refactors drag-drop to typescript
Refactors js state to ts slice
Updates types
Moves useThrottledCallback and useDebouncedCallback to common
Improves perf by removing extra code and callbacking things

also reverts locks back

### CODE REVIEW

File: packages/web/src/components/nav/desktop/PlaylistLibrary.tsx [hunk 1]
```diff
@@ -215,7 +214,7 @@ const PlaylistLibrary = ({
       record(
         make(Name.PLAYLIST_LIBRARY_REORDER, {
           containsTemporaryPlaylists: containsTempPlaylist(newLibrary),
-          kind: draggingKind
+          kind: draggingKind as any
```
Comment (sliptype) [inline packages/web/src/components/nav/desktop/PlaylistLibrary.tsx:217]: 👀

**Review by sliptype — APPROVED** [review]
Cleannnnn

File: packages/web/src/components/nav/desktop/PlaylistLibrary.tsx (hunk 1 above)
Comment (dylanjeffers) [reply packages/web/src/components/nav/desktop/PlaylistLibrary.tsx]: fixed :)

**Comment by audius-infra** [comment]
It looks like there may be some changes to native mobile code, which requires triggering a full app release. Please follow the instructions here: https://www.notion.so/audiusproject/When-to-bump-app-version-2644a8f772364a4d91f44abcba44ce0b?pvs=4. cc @nicoback2 @sliptype

**Pull request merged**